### PR TITLE
Add error model and use it as response schema for 4xx and 5xx responses

### PR DESCRIPTION
**Pull request by emi80**

### CODE REVIEW

File: rnaget-openapi.yaml [hunk 1]
```diff
@@ -587,6 +673,14 @@ definitions:
     externalDocs:
       description: "Find out more"
       url: "https://github.com/ga4gh-rnaseq/schema"
+  error:
+    type: "object"
+    description: "General API error model"
+    properties:
+      code:
```
Comment (saupchurch) [inline rnaget-openapi.yaml]: Do we need this as a property? The response code should be available directly in the response object.

**Comment by jb-adams** [comment]
Should we also include in `rnaget.md` a short explanation that all responses are expected to be in JSON?

File: rnaget-openapi.yaml (hunk 1 above)
Comment (emi80) [reply rnaget-openapi.yaml]: I don't think it's a required property. I added it as it seemed convenient to me.
I can remove it if you prefer to have a simpler error model.

**Comment by emi80** [comment]
> Should we also include in `rnaget.md` a short explanation that all responses are expected to be in JSON, regardless of status code?

I think so

**Comment by saupchurch** [comment]
@emi80 I agree - can you add that note to the main spec?

**Comment by emi80** [comment]
Done!

File: rnaget-openapi.yaml (hunk 1 above)
Comment (emi80) [reply rnaget-openapi.yaml]: Any though on this?

Comment (saupchurch) [reply rnaget-openapi.yaml]: I think we should keep things as simple as possible - we can add fields later if a use case arises

Comment (emi80) [reply rnaget-openapi.yaml]: Sounds good. I just removed the `code` property.

**Comment by saupchurch** [comment]
looks good merging

**Pull request merged**